third_party/Makefile

```diff
@@ -0,0 +1,20 @@
+.PHONY: generate-third-party-deps
+generate-third-party-deps:
+	git clone git@github.com:go-enry/go-oniguruma.git
```

Comment:
I wouldn't clone the Git repo inside the parent repo like this; you're removing the .git and .github directories, but that's going to be very fragile.

If you want to offer this, I would put it in a temporary directory.
```
third_party/go-oniguruma:
  DIR=$$(mktemp -d) \
  cd "$DIR" && \
  git clone --branch v1.2.1 git@github.com:go-enry/go-oniguruma.git go-oniguruma && \
  cd go-oniguruma && \
  git apply $(abspath "go-oniguruma.patch") && \
  rm -r "$(abspath $@)" && \
  cp -a "$DIR/go-oniguruma" "$(abspath $@)"
```

(and repeat with a separate rule for oniguruma.git)

Reply:
Thank you for this suggestion! This is better, though i'm still curious why cloning in a temp folder and then copying works better. It does work without having to remove the `.git` folder, but why does this work ? 🤔 

I also created `update_go-oniguruma.patch` and `update_oniguruma.patch` rules with this setup.

Reply:
`git`'s behavior inside a repository is partially controlled by that repository. Cloning in a temporary directory ensures that `git` is executing from a clean state, and neither repository is polluting the other.

Reply:
Thanks for the explanation!